cmake/variables.cmake

```
@@ -153,13 +153,33 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
   list(APPEND ML_COMPILE_DEFINITIONS BOOST_ALL_NO_LIB)
 endif()
 
-if(UNIX AND "$ENV{ML_DEBUG}" AND "$ENV{ML_COVERAGE}")
+# Perform a "Release" build by default...
+if(NOT CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE Release)
+endif()
+
+message(STATUS "CMAKE_BUILD_TYPE: ${CMAKE_BUILD_TYPE}")
+
+# However, to keep in step with our historical
+# build system if ML_DEBUG is set to a boolean
+# true value, e.g. ML_DEBUG=1, then override
+# the build type to be Debug
+if(DEFINED ENV{ML_DEBUG} AND $ENV{ML_DEBUG})
```

This causes a syntax error if `$ENV{ML_DEBUG}` evaluates to nothing:

```
CMake Error at cmake/variables.cmake:167 (if):
  if given arguments:

    "DEFINED" "ENV{ML_DEBUG}" "AND"

  Unknown arguments specified
```

Since an empty string evaluates to false I think this is the way to do it:

```suggestion
if("$ENV{ML_DEBUG}")
```